pkgs/os-specific/linux/freeipa/default.nix

```
@@ -0,0 +1,97 @@
+{ stdenv, lib, fetchurl, pkgconfig, autoconf, automake, kerberos,
+  openldap, popt, sasl, curl, xmlrpc_c, ding-libs, p11-kit, gettext, nspr, nss,
+  _389-ds-base, svrcore, libuuid, talloc, tevent, samba, libunistring, libverto,
+  libpwquality, systemd, python3, bind, sssd, jre, rhino, lesscpy, jansson,
+  runtimeShell }:
+
+let
+  pathsPy = ./paths.py;
+
+  pythonInputs = with python3.pkgs; [
+    six ldap dns netaddr netifaces gssapi dogtag-pki pyasn1 sssd cffi lxml
+    dbus-python cryptography memcached qrcode pyusb yubico setuptools jinja2
+    augeas samba
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "freeipa";
+  version = "4.10.1";
+
+  src = fetchurl {
+    url = "https://releases.pagure.org/freeipa/freeipa-${version}.tar.gz";
+    sha256 = "sha256-q2rQzcBl1tI4/7+hxEwOY9ND86hObe7O7Y9EEH7cUoA=";
+  };
+
+  nativeBuildInputs = [
+    python3.pkgs.wrapPython jre rhino lesscpy automake autoconf gettext
+  ];
+
+  buildInputs = [
+    kerberos openldap popt sasl curl xmlrpc_c pkgconfig ding-libs p11-kit
+    python3 autoconf nspr nss _389-ds-base svrcore libuuid talloc tevent samba
+    libunistring libverto systemd bind libpwquality jansson
+  ] ++ pythonInputs;
+
+  postPatch = ''
+    patchShebangs makeapi makeaci install/ui/util
+
+    substituteInPlace ipaplatform/setup.py \
+      --replace 'ipaplatform.debian' 'ipaplatform.nixos'
+
+    substituteInPlace ipasetup.py.in \
+      --replace 'int(v)' 'int(v.replace("post", ""))'
+
+    substituteInPlace client/ipa-join.c \
+      --replace /usr/sbin/ipa-getkeytab $out/bin/ipa-getkeytab
+
+    cp -r ipaplatform/{fedora,nixos}
+    substitute ${pathsPy} ipaplatform/nixos/paths.py \
+      --subst-var out \
+      --subst-var-by bind ${bind.dnsutils} \
+      --subst-var-by curl ${curl} \
+      --subst-var-by kerberos ${kerberos}
+  '';
+
+  PYTHON = "${python3}/bin/python";
+
+  NIX_CFLAGS_COMPILE = "-I${_389-ds-base}/include/dirsrv";
+  pythonPath = pythonInputs;
+
+  # Building and installing the server fails with silent Rhino errors, skipping
+  # for now. Need a newer Rhino version.
+  #buildFlags = [ "client" "server" ]
+
+  configureFlags = [
+   "--with-systemdsystemunitdir=$out/lib/systemd/system"
+   "--with-ipaplatform=nixos"
+   "--disable-server"
+  ];
```

```suggestion
  configureFlags = [
    "--with-systemdsystemunitdir=$out/lib/systemd/system"
    "--with-ipaplatform=nixos"
    "--disable-server"
  ];
```
try something like nixpkgs-fmt :)

---

Sure. Sorry about that. Will apply